### PR TITLE
Implement IEC 61853 module temperature model

### LINKED ISSUE #750: Implement IEC 61853 module temperature model
The Faiman module temperature model is used in IEC 61853.  It conceptually not much different from others, but uses slightly different parameters.

### PR DESCRIPTION
**Pull request by adriesse**
- [x] Closes #750
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - [x] Updates entries to [`docs/sphinx/source/api.rst`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/api.rst) for API changes.
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

**Comment by adriesse** [comment]
A quick first look at the function signature and 6 lines of code would be helpful before I flesh out the tests.

**Comment by cwhanse** [comment]
Thanks for this.  The signature looks fine to me. Maybe `U0=25.0` and `U1=6.84` as defaults, those values appear in Faiman's paper from measured temperature for 7 silicon modules.  I didn't see any default values in the 2016 version of IEC61853-2.

Maybe combine a few of the calculation lines, a style preference more than anything else.

**Comment by adriesse** [comment]
> Maybe combine a few of the calculation lines, a style preference more than anything else.

Not really my usual style either, but it's intended to help people understand the model.

**Comment by adriesse** [comment]
Ok, ready for full review.

File: pvlib/temperature.py [hunk 1]
```diff
@@ -269,3 +269,69 @@ def pvsyst_cell(poa_global, temp_air, wind_speed=1.0, u_c=29.0, u_v=0.0,
     heat_input = poa_global * alpha_absorption * (1 - eta_m)
     temp_difference = heat_input / total_loss_factor
     return temp_air + temp_difference
+
+
+def faiman(poa_global, temp_air, wind_speed=1.0, u0=25.0, u1=6.84):
+    '''
+    Calculate cell/module temperature using an empirical heat loss factor
```
Comment (cwhanse) [inline pvlib/temperature.py]: would prefer "cell or module" to avoid suggesting the output is a ratio

File: pvlib/temperature.py [hunk 2]
```diff
@@ -269,3 +269,69 @@ def pvsyst_cell(poa_global, temp_air, wind_speed=1.0, u_c=29.0, u_v=0.0,
...
+        Combined heat loss factor influenced by wind. The default value is one
+        determined by Faiman for 7 silicon modules. [(W/m^2 C)(m/s)].
+
+    Returns
+    -------
+    numeric, values in degrees Celsius
+
+    Notes
+    -----
+    All arguments may be scalars or vectors. If multiple arguments
```
Comment (cwhanse) [inline pvlib/temperature.py]: The Note doesn't hurt but I don't think it's needed.

**Comment by kevinsa5** [comment]
One note, the thermal coefficients are called `u0`, and `u1` in the signature but `u_c` and `u_v` in the docstring.

File: pvlib/temperature.py (hunk 2 above)
Comment (adriesse) [reply pvlib/temperature.py]: I also considered about using `array-like`, but that depends on the conclusion and transition plan for #765.  (Only one definition of 'array-like' should apply at one time.)

**Comment by cwhanse** [comment]
@adriesse I fixed the merge conflict in whatsnew.

**Comment by CameronTStark** [comment]
> > Maybe combine a few of the calculation lines, a style preference more than anything else.
> 
> Not really my usual style either, but it's intended to help people understand the model, in particular the difference with the PVsyst model.

I really appreciate when math functions are coded like this by breaking out sections of the equation and providing a descriptive variable name for each section and cleanly arranging the pieces afterward. It strikes me as very Pythonic in that it creates more readable code and is therefore more easily debugged all while educating the reader and being computationally functional. A win-win-win in my opinion.

**Comment by wholmgren** [comment]
Thanks @adriesse!

**Comment by adriesse** [comment]
Thanks @all!

**Pull request merged**